ChaosMod/Components/EffectDispatcher.cpp

```diff
@@ -566,6 +574,18 @@ void EffectDispatcher::ClearMostRecentEffect()
 	}
 }
 
+std::vector<RegisteredEffect*> EffectDispatcher::GetRecentEffects(int distance) const
+{
+	std::vector<RegisteredEffect*> temp;
+	for (int i = m_rgDispatchedEffectsLog.size() - 1; i > distance && i > 0; i--)
+	{
+		RegisteredEffect* regeff = m_rgDispatchedEffectsLog.at(i);
+		if (std::count(temp.begin(), temp.end(), regeff)) continue;
```

Comment:
Use `std::find` instead (more efficient).

Additionally, if a duplicate effect was found it will still decrement `i`, meaning the amount of returned effects != `distance`. Instead of checking if `i` reached `distance` in the for loop declaration, it might be wiser to manually keep track of the amount (e.g. checking if it added an effect to `temp` and decrementing `distance` if so, until `distance` reaches <= 0)